### PR TITLE
Fixed short vector casts inside template

### LINKED ISSUE #3073: Syntax error for casts to pointers to short vectors inside templates
ISPC reports a syntax error for casts to pointers to short vectors inside templates.

```ispc
template <typename T>
unmasked void foo(uniform int8 *uniform a) {
    uniform T<4>*uniform ptr1 = (uniform T<4> *uniform)a;
    T<4> *ptr2 = (T<4> *)a;
}
```

Output:
```bash
<source>:3:44: Error: syntax error, unexpected '<', expecting ')'. 
    uniform T<4> *uniform ptr2 = (uniform T<4> *uniform)a;
                                           ^
Compiler returned: 1
```

Compiler explorer link: https://godbolt.org/z/MbE7739GK

### PR DESCRIPTION
**Pull request by aneshlya**
Fixes #3073

### CODE REVIEW

**Review by nurmukhametov — APPROVED** [review]
LGTM!

File: tests/lit-tests/func_template_shortvec_10.ispc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// CHECK: (instantiation <varying int32>) {{.*}}  [ unmasked varying int32<4>(uniform int8 * uniform a)] "foo1"
+// CHECK: TypeCastExpr {{.*}} [varying int32<4> * varying]
+
+// CHECK: unmasked /*unbound*/ T<N>(uniform int8 * uniform a)] "foo2"
+// CHECK: TypeCastExpr {{.*}} [/*unbound*/ T<N> * /*unbound*/]
+// CHECK: (instantiation <varying int32, 4>) {{.*}}  [ unmasked varying int32<4>(uniform int8 * uniform a)] "foo2"
+// CHECK: TypeCastExpr {{.*}} [varying int32<4> * varying]
+template <typename T> noinline unmasked T<4> foo1(uniform int8 *uniform a) {
+    uniform T<4> *uniform ptr1 = (uniform int<4> *uniform)a;
+    T<4> *ptr2 = (T<4> *)a;
```
Comment (dbabokin) [inline tests/lit-tests/func_template_shortvec_10.ispc:14]: The template function is declared to return `unmasked T<4>` value, but has not return statement... Is it intended?

**Review by dbabokin — APPROVED** [review]
LGTM

File: tests/lit-tests/func_template_shortvec_10.ispc (hunk 1 above)
Comment (aneshlya) [reply tests/lit-tests/func_template_shortvec_10.ispc:14]: No, it wasn't intentional, I fixed it.

**Pull request merged**